cli/determined_cli/trial.py

```diff
@@ -82,25 +82,38 @@ def describe_trial(args: Namespace) -> None:
 
 @authentication_required
 def logs(args: Namespace) -> None:
-    offset, state = 0, None
+    last_offset, last_state = 0, None
 
-    def print_logs(limit: Optional[int] = None) -> None:
-        nonlocal offset, state
-        path = "trials/{}/logsv2?offset={}".format(args.trial_id, offset)
+    def print_logs(offset, limit: Optional[int] = 5000) -> int:
```

Comment:
blocking: `last_offset` needs to always be updated when this function is called, or `--follow` is completely broken. Would it be workable to mostly go back to the old way of using the external variables directly and setting them in here, perhaps with the initial value of `last_offset` at `None` instead? (That would also allow you to undo the addition of `last_` to those names.)

Reply:
yup, I needed to fix that. IMO keeping the inputs to the function the only things that determine its behavior makes it easier to reason about and hopefully less error prone.